File: src/main/java/sunshine/infrastructure/openmeteo/OpenMeteoResponse.java
package sunshine.infrastructure.openmeteo;

import com.fasterxml.jackson.annotation.JsonProperty;

public record OpenMeteoResponse(Current current) {
    public record Current(
        @JsonProperty("temperature_2m") double temperature2m,
        @JsonProperty("relative_humidity_2m") int relativeHumidity2m,
        @JsonProperty("apparent_temperature") double apparentTemperature,
        @JsonProperty("weather_code") int weatherCode,
        @JsonProperty("wind_speed_10m") double windSpeed10m
    ) {}
}

File: src/main/kotlin/sunshine/infrastructure/openmeteo/OpenMeteoResponse.kt
package sunshine.infrastructure.openmeteo

import com.fasterxml.jackson.annotation.JsonProperty

/**
 * OpenMeteo API 응답을 매핑하는 DTO.
 */
data class OpenMeteoResponse(
    val current: Current
) {
    data class Current(
        @JsonProperty("temperature_2m")
        val temperature2m: Double,
        @JsonProperty("relative_humidity_2m")
        val relativeHumidity2m: Int,
        @JsonProperty("apparent_temperature")
        val apparentTemperature: Double,
        @JsonProperty("weather_code")
        val weatherCode: Int,
        @JsonProperty("wind_speed_10m")
        val windSpeed10m: Double
    )
}
